death_note.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import "sync"
+
+type DeathNote struct {
+	paramSet map[string]bool
+	rwMutex sync.RWMutex
+}
+
+func NewDeathNote() *DeathNote {
+	return &DeathNote{
+		paramSet: map[string]bool{},
+	}
+}
+
+func (d *DeathNote) AddParam(param string) {
```

Comment:
```suggestion
func (dn *DeathNote) AddParam(param string) {
```

completely nit: usually better to use both letter from camel cased type, as it maintains the mind context, easing the read process